cli/commands/actions_use_codegen.go

```diff
@@ -0,0 +1,206 @@
+			if ec.Config.Version == "1" {
+				return fmt.Errorf("actions commands can be executed only when config version is greater than 1")
+			}
+			if ec.MetadataDir == "" {
+				return fmt.Errorf("actions commands can be executed only when metadata_dir is set in config")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return opts.run()
+		},
+	}
+
+	f := actionsUseCodegenCmd.Flags()
+
+	f.StringVar(&opts.framework, "framework", "", "")
+	f.StringVar(&opts.outputDir, "output-dir", "", "")
+	f.BoolVar(&opts.withStarterKit, "with-starter-kit", false, "")
+
+	f.String("endpoint", "", "http(s) endpoint for Hasura GraphQL Engine")
+	f.String("admin-secret", "", "admin secret for Hasura GraphQL Engine")
+	f.String("access-key", "", "access key for Hasura GraphQL Engine")
+	f.MarkDeprecated("access-key", "use --admin-secret instead")
+
+	// need to create a new viper because https://github.com/spf13/viper/issues/233
+	v.BindPFlag("endpoint", f.Lookup("endpoint"))
+	v.BindPFlag("admin_secret", f.Lookup("admin-secret"))
+	v.BindPFlag("access_key", f.Lookup("access-key"))
+
+	return actionsUseCodegenCmd
+}
+
+type actionsUseCodegenOptions struct {
+	EC *cli.ExecutionContext
+
+	framework      string
+	outputDir      string
+	withStarterKit bool
+}
+
+func (o *actionsUseCodegenOptions) run() (err error) {
+	// ensure the the actions-codegen repo is cloned and updated
+
+	o.framework = strings.TrimSpace(o.framework)
+	o.outputDir = strings.TrimSpace(o.outputDir)
+
+	o.EC.Spin("Fetching codegen assets...")
+	defer o.EC.Spinner.Stop()
+	actionsCodegenGit := util.NewGitUtil(
+		actionsCodegenRepoURI,
+		filepath.Join(o.EC.GlobalConfigDir, actionsCodegenDirName),
+		"",
+	)
+	err = actionsCodegenGit.EnsureCloned()
+	if err != nil {
+		return
+	}
+	_ = actionsCodegenGit.EnsureUpdated()
+	newCodegenExecutionConfig := o.EC.Config.Action.Codegen
+	o.EC.Spinner.Stop()
+
+	frameworkFileBytes, err := ioutil.ReadFile(filepath.Join(o.EC.GlobalConfigDir, actionsCodegenDirName, "frameworks.json"))
+	if err != nil {
+		return err
+	}
+	var allFrameworks []codegenFramework
+	err = json.Unmarshal(frameworkFileBytes, &allFrameworks)
+	if err != nil {
+		return err
+	}
+
+	// if framework flag is not provided, display a list and allow them to choose
+	if o.framework == "" {
+		var frameworkList []string
+		for _, f := range allFrameworks {
+			frameworkList = append(frameworkList, f.Name)
+		}
+		sort.Strings(frameworkList)
+		newCodegenExecutionConfig.Framework, err = util.GetSelectPrompt("Choose a codegen framework to use", frameworkList)
+		if err != nil {
+			return err
+		}
+	} else {
+		newCodegenExecutionConfig.Framework = o.framework
+	}
+
+	hasStarterKit := false
+	for _, f := range allFrameworks {
+		if f.Name == newCodegenExecutionConfig.Framework && f.HasStarterKit {
+			hasStarterKit = true
+		}
+	}
+
+	// if with-starter-kit flag is not provided, give an option to clone a starterkit
+	if !o.withStarterKit && hasStarterKit {
```

Comment:
If with-starter-kit flag is set, but the starter kit was not found, command should fail